### PR TITLE
python3Packages.mizani: add missing dependency

### PR DESCRIPTION
**Pull request by fabaff**
###### Motivation for this change
Fix build (https://hydra.nixos.org/build/162222402)

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 153096` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python38Packages.mizani</li>
    <li>python39Packages.mizani</li>
  </ul>
</details>

File: pkgs/development/python-modules/mizani/default.nix [hunk 1]
```diff
@@ -1,15 +1,20 @@
...
+{ lib
+, buildPythonPackage
 , fetchFromGitHub
-, lib
 , matplotlib
 , palettable
 , pandas
 , pytestCheckHook
+, pythonOlder
+, scipy
```
Comment (samuela) [inline pkgs/development/python-modules/mizani/default.nix:9]: nit: alphabetize

File: pkgs/development/python-modules/mizani/default.nix [hunk 2]
```diff
@@ -1,15 +1,20 @@
...
 }:
 
 buildPythonPackage rec {
   pname = "mizani";
   version = "0.7.3";
+  format = "setuptools";
```
Comment (samuela) [inline pkgs/development/python-modules/mizani/default.nix:15]: Why is this required now? This wasn't necessary when I originally packaged it, and it doesn't look like `src` has changed.

**Review by samuela — CHANGES_REQUESTED** [review]
Why is this breaking now? `src` hasn't changed and the tests all passed when I packaged it originally. These changes LGTM except I don't understand why it's only breaking now.

File: pkgs/development/python-modules/mizani/default.nix (hunk 1 above)
Comment (fabaff) [reply pkgs/development/python-modules/mizani/default.nix:9]: `lib` and `stdenv` are usually (or often) mentioned first. I can revert that as it's the primary maintainer's call.

File: pkgs/development/python-modules/mizani/default.nix (hunk 2 above)
Comment (fabaff) [reply pkgs/development/python-modules/mizani/default.nix:15]: The way how Python packages are build will change in the near future. For now it's more of a "marker" and I add them when I have to edit the files.

**Comment by fabaff** [comment]
> Why is this breaking now? src hasn't changed and the tests all passed when I packaged it originally. These changes LGTM except I don't understand why it's only breaking now.

In the past, `scipy` was pulled in by `pandas` I assume. Upstream added [it](https://github.com/has2k1/mizani/blob/master/setup.py#L40) in the meantime to make sure that `scipy` is available during runtime.

**Comment by samuela** [comment]
Ok, that makes sense to me. Thanks for fixing this!
